### ISSUE TITLE
Text/Math Recognition

### ISSUE THREAD

**Issue by OmegaRogue** [flxzt/rnote#327]
**Is your feature request related to a problem? Please describe.**
Taking handwritten notes is useful for quickly writing down things with sketches or math without the need for a keyboard, as a replacement for taking notes on Paper, but having digital Text is useful for searching through it and other use cases.
**Describe the solution you'd like**
Having the option to select handwritten text and convert it to a text box, and the option to select formulas other math to convert it to a formatted (maybe LaTeX) math element, like Xournal++ supports with LaTeX integration.

**Additional context**
OneNote supports OCR for mathematical formulas and equations

**Comment by PhilDevProg** [comment]
For text recognition, [Tesseract](https://github.com/tesseract-ocr/tesseract) could be used. Here are the Rust bindings: https://github.com/antimatter15/tesseract-rs
I also found this: https://pyimagesearch.com/2020/08/24/ocr-handwriting-recognition-with-opencv-keras-and-tensorflow/ which describes how to do handwritten text recognition by using OpenCV, Keras & TensorFlow using Python but the OCR tool would have to be trained by the user (but this would propably result in better recognition).

**Comment by ashrude** [comment]
The process described in there would technically work, but since we'd have access to the stroke information, it could be implemented much better. Having stroke information allows for higher accuracy.

**Comment by Doublonmousse** [comment]
For the math conversion, this looks more like Task 1 of the CROHME competition https://crohme2023.ltu-ai.dev/data-tools/
This uses inkml as the file format for the stroke data and uses the stroke information ((x,y) position only).

Ideally, this would work in an incremental fashion and update the prediction for each new stroke rather than (or in addition to) only be accessible as a tool that convert a stroke selection to the math equivalent.

Conversion of a selection is only good the conversion is very good. If not you end up trying it a couple time, see that it fails to convert moderately complex equations and end up not using it (all the more so if there is no easy way to modify the converted equation afterward).

 Though #788 would have to be completed before this can be tackled

**Comment by PellelNitram** [comment]
Hi! I’m working on [Xournal++ HTR](https://github.com/PellelNitram/xournalpp_htr), a local [handwritten text recognition](https://en.wikipedia.org/wiki/Handwriting_recognition) plugin for Xournal++. I figured commenting here might help with this issue.

The plugin processes Xournal++ documents and makes them searchable by converting handwritten text into actual text. When exported as a PDF, the recognized text is added as an invisible layer, enabling full-text search; here’s a short [demo video](https://www.youtube.com/watch?v=rwHbacjjAS8) and see this image of an example conversion.

<img width="750" height="487" alt="Image" src="https://github.com/user-attachments/assets/0ea10597-7b45-4bd9-9ee9-f6d86bbfdaf3" />

The project is still in development, and while the quality isn’t perfect yet, it’s already usable. We also have an [online demo](https://huggingface.co/spaces/PellelNitram/xournalpp_htr) that works without any installation. Feel free to try it out with a Xournal++ file! :-)

We currently have an open [issue](https://github.com/PellelNitram/xournalpp_htr/issues/36) about adding handwritten text recognition support for rnote files, and I wanted to briefly check whether there’s interest in that here as well.

Cheers, and enjoy the demo! :-)

Happy to answer any question!
